worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java

```diff
@@ -3298,7 +3298,7 @@ public int hollowOutRegion(Region region, int thickness, Pattern pattern, Mask m
                 outer:
                 for (BlockVector3 position : region) {
                     for (BlockVector3 recurseDirection : recurseDirections) {
-                        BlockVector3 neighbor = position.add(recurseDirection);
+                        BlockVector3 neighbor = position.toImmutable().add(recurseDirection);
```

Comment:
Nice find. I would say to use the mutable bv3 as is done in the code above to avoid creating many BV3s here though